### PR TITLE
[release/6.0.1xx] Bump System.Security.Cryptography.Xml to 6.0.1 and System.Security.Cryptography.Pkcs to 6.0.4

### PR DESCRIPTION
**Pull request by YuliiaKovalova**
### Summary
This is a security fix after the recent scanner results for sdk repo.

### Customer Impact
No impact is expected.

### Regression?
No

### Testing
N/A

### Risk
Minimal - no risks, it's just package version update + data from arcade .Net 6 Eng channel

### CODE REVIEW

**Review by rokonec — APPROVED** [review]

**Comment by YuliiaKovalova** [comment]
Hi @premun,

Could you help us with that? We need a build with the dependency bump .

**Comment by premun** [comment]
Sorry, I don't follow. Help with what exactly?

**Comment by YuliiaKovalova** [comment]
> Sorry, I don't follow. Help with what exactly?

@premun , I apologize for confusion.
This build fails on source-build step and I don't know how to propagate the arcade version that can fix it and don't break sdk.
Do you have any ideas?

**Comment by premun** [comment]
Sorry, I am not the best person for SB problems. @dotnet/source-build-internal Yulia has some issues with restoring `Microsoft.DotNet.SourceBuild.Tasks`. Can you help?

**Comment by mthalman** [comment]
```
##[error].packages/microsoft.dotnet.arcade.sdk/6.0.0-beta.21460.7/tools/Tools.proj(0,0): error NU1603: (NETCORE_ENGINEERING_TELEMETRY=Restore) Tools depends on Microsoft.DotNet.SourceBuild.Tasks (>= 6.0.0-beta.21460.7) but Microsoft.DotNet.SourceBuild.Tasks 6.0.0-beta.21460.7 was not found. An approximate best match of Microsoft.DotNet.SourceBuild.Tasks 6.0.0-beta.21620.3 was resolved.
  /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/6.0.0-beta.21460.7/tools/Tools.proj : error NU1102: Unable to find package Microsoft.SourceBuild.Intermediate.arcade with version (= 6.0.0-beta.21460.7)
  /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/6.0.0-beta.21460.7/tools/Tools.proj : error NU1102:   - Found 1002 version(s) in arcade [ Nearest version: 6.0.0-beta.22431.8 ]
  /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/6.0.0-beta.21460.7/tools/Tools.proj : error NU1102:   - Found 0 version(s) in dotnet-public
  /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/6.0.0-beta.21460.7/tools/Tools.proj : error NU1102:   - Found 0 version(s) in dotnet-tools
  /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/6.0.0-beta.21460.7/tools/Tools.proj : error NU1102:   - Found 0 version(s) in dotnet6
##[error].packages/microsoft.dotnet.arcade.sdk/6.0.0-beta.21460.7/tools/Tools.proj(0,0): error NU1102: (NETCORE_ENGINEERING_TELEMETRY=Restore) Unable to find package Microsoft.SourceBuild.Intermediate.arcade with version (= 6.0.0-beta.21460.7)
  - Found 1002 version(s) in arcade [ Nearest version: 6.0.0-beta.22431.8 ]
  - Found 0 version(s) in dotnet-public
  - Found 0 version(s) in dotnet-tools
  - Found 0 version(s) in dotnet6
    Failed to restore /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/6.0.0-beta.21460.7/tools/Tools.proj (in 968 ms).
```

The error seems completely unrelated to the changes. Are you sure this branch is building without these changes?

Do you have the latest 6.0 Arcade in this branch? /cc @mmitche

**Comment by premun** [comment]
There was no commit on the target branch for some time:

image

File: global.json [hunk 1]
```diff
@@ -15,6 +15,6 @@
   },
   "msbuild-sdks": {
     "Microsoft.Build.CentralPackageVersions": "2.0.1",
-    "Microsoft.DotNet.Arcade.Sdk": "6.0.0-beta.21460.7"
+    "Microsoft.DotNet.Arcade.Sdk": "6.0.0-beta.21620.3"
```
Comment (MichaelSimons) [inline global.json:18]: You need to also update the version.details.xml file.

**Comment by YuliiaKovalova** [comment]
@GangWang01 please look at the tests failures

File: eng/Versions.props [hunk 2]
```diff
@@ -2,7 +2,8 @@
 <!-- Copyright (c) .NET Foundation and contributors. All rights reserved. Licensed under the MIT license. See License.txt in the project root for full license information. -->
 <Project>
   <PropertyGroup>
-    <VersionPrefix>17.0.1</VersionPrefix><DotNetFinalVersionKind>release</DotNetFinalVersionKind>
+    <VersionPrefix>17.0.2</VersionPrefix>
+    <DotNetFinalVersionKind>release</DotNetFinalVersionKind>
```
Comment (rainersigwald) [inline eng/Versions.props:6]:
```suggestion
    <VersionPrefix>17.0.2</VersionPrefix><DotNetFinalVersionKind>release</DotNetFinalVersionKind>
```

File: src/Build/Properties/launchSettings.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "profiles": {
+    "BuilderApp": {
+      "commandName": "Executable",
+      "executablePath": "C:\\msbuild\\locator\\ykovalova\\MSBuildLocator\\samples\\BuilderApp\\bin\\Debug\\net472\\BuilderApp.exe",
```
Comment (rainersigwald) [inline src/Build/Properties/launchSettings.json:5]: Better remove this file.

**Comment by GangWang01** [comment]
> @GangWang01 please look at the tests failures.
> 
> update: it's related to vs-image update. Hopefully, @rainersigwald will resolve it.

Thanks Rainer 

> @GangWang01 please look at the tests failures.
> 
> update: it's related to vs-image update. Hopefully, @rainersigwald will resolve it.

It looks like the test failures were resolved by Rainer. Thanks Rainer!

**Review by JanKrivanek — APPROVED** [review]
I suppose changes in eng\.. were automatic via the arcade update - correct? Or is there any specific manual changes that should be reviewed with care?

**Comment by YuliiaKovalova** [comment]
> I suppose changes in eng.. were automatic via the arcade update - correct? Or is there any specific manual changes that should be reviewed with care?

Nope, these changes were added by running
`darc update-dependencies --source-repo "dotnet/arcade" --channel ".NET 6 Eng"`

**Comment by KevinRansom** [comment]
@YuliiaKovalova will you be merging this anytime soon.  I'm hoping for F# to use the nuget packages from it.

File: eng/Packages.props [hunk 4]
```diff
@@ -31,8 +31,8 @@
     <PackageReference Update="System.Reflection.Metadata" Version="1.6.0" />
     <PackageReference Update="System.Resources.Extensions" Version="$(SystemResourcesExtensionsPackageVersion)" />
     <PackageReference Update="System.Runtime.CompilerServices.Unsafe" Version="5.0.0"/>
-    <PackageReference Update="System.Security.Cryptography.Pkcs" Version="4.7.0" />
-    <PackageReference Update="System.Security.Cryptography.Xml" Version="4.7.0" />
+    <PackageReference Update="System.Security.Cryptography.Pkcs" Version="6.0.1" />
```
Comment (marcpopMSFT) [inline eng/Packages.props:34]: should be version 6.0.4 of Pkcs

**Pull request merged**